frontend/src/pages/user-profile/components/user-profile-sidebar/user-profile-sidebar.tsx

```diff
@@ -51,6 +66,46 @@ const UserProfileSidebar: React.FC<Properties> = ({
     void dispatch(authActions.signOut());
   }, [dispatch]);
 
+  const handleCloseModal = useCallback((): void => {
+    setModalOpen(false);
```

Comment:
I think there's boolean that describes current dialog behaviour, as I remember `ref.current.open`. Try to find out and implement it, then you don't have to use `onClose` prop and state

Reply:
@Dmytro-Kucherenko is right. Please check our latest flow with modal